src/app/shared/example-viewer/example-code-viewer.ts

```diff
@@ -0,0 +1,37 @@
+import {Component, EventEmitter, Input, Output} from '@angular/core';
+
+@Component({
+  selector: 'example-code-viewer',
+  templateUrl: './example-code-viewer.html',
+  styleUrls: ['./example-viewer.scss'],
+})
+export class ExampleCodeViewer {
+  /** Map of example files that should be displayed in the view-source tab. */
+  @Input() exampleTabs: {[tabName: string]: string};
+
+  /** Name of file to display. */
+  @Input() fileName: string;
+
+  /** Whether the component is in full or compact view. (Whether to show copy button.) */
+  @Input() compactView: boolean;
+
+  /** Range of lines of the source code to display - only valid in compact view. */
+  @Input() lines: [number, number];
+
+  /** Whether to show toggle for compact view. */
+  @Input() showCompactToggle: boolean;
+
+  @Output() toggleCompactViewEvent = new EventEmitter();
+
+  @Output() copySourceEvent = new EventEmitter();
```

Comment:
maybe just `copySource`? I don't particularly like the `Event` because it's kind of implied (like the old `iLineNumber`/`bShowCompactToggle` convention). But this is just an opinion so take it or leave it.